internal/provider/resource_status_page_group.go

```diff
@@ -0,0 +1,151 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"reflect"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+var statusPageGroupSchema = map[string]*schema.Schema{
+	"team_name": {
+		Description: "Used to specify the team the resource should be created in when using global tokens.",
+		Type:        schema.TypeString,
+		Optional:    true,
+		Default:     nil,
+		DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+			return d.Id() != ""
+		},
+	},
+	"id": {
+		Description: "The ID of this status page group.",
+		Type:        schema.TypeString,
+		Optional:    false,
+		Computed:    true,
+	},
+	"name": {
+		Description: "A name of the group that you can see in the dashboard.",
+		Type:        schema.TypeString,
+		Optional:    true,
+		Computed:    true,
```

Comment:
I couldn't find the API docs near https://betterstack.com/docs/uptime/api/list-all-existing-status-pages/, but I have a feeling the name should be required, shouldn't it?

```suggestion
		Required:    true,
```

Reply:
Good catch